infra/test/integration/testhelper.go

```diff
@@ -146,9 +151,26 @@ func assertResponseContains(t *testing.T, assert *assert.Assertions, url string,
 	assert.Nil(err)
 	assert.GreaterOrEqual(code, 200)
 	assert.LessOrEqual(code, 299)
+
+	// Assert that all fragments are present in responseBody
+	if !oneOf {
+		for _, fragment := range text {
+			assert.Containsf(responseBody, fragment, "couldn't find %q in response body", fragment)
+		}
+		return
+	}
+
+	// Otherwise, assert that at minimum one fragment is present in responseBody
 	for _, fragment := range text {
-		assert.Containsf(responseBody, fragment, "couldn't find %q in response body", fragment)
+		var hasFragment bool = strings.Contains(responseBody, fragment)
+
+		if hasFragment {
+			assert.Truef(hasFragment, "verified %q is in response body", fragment)
```

Comment:
nit: this line will never through assert. it can be removed safely.

Reply:
This runs when `oneOf` bool flag is true and that (at minimum) one of the strings that has been passed is present in the final `responseBody`. Check out build logs https://pantheon.corp.google.com/cloud-build/builds/87736e79-f796-433f-a2f7-9de4df7daa49.